remixer_core/src/main/java/com/google/android/libraries/remixer/Remixer.java

```diff
@@ -53,12 +60,27 @@ public static Remixer getInstance() {
     return instance;
   }
 
-  Remixer() {
+  /**
+   * Visible only for testing. Users should only use {@link #getInstance()}.
+   */
+  public Remixer() {
     keyMap = new HashMap<>();
     remixerItems = new ArrayList<>();
   }
 
   /**
+   * Set the synchronization mechanism to keep variables in sync locally and (possibly) with
+   * external sources.
+   *
+   * <p>Remixer relies on a SynchronizationMechanism instance to be the source of truth of the
+   * values and configuration, so the user should always set a SynchronizationMechanism.
```

Comment:
it's not complete clear from this javadoc, but does the user really have to manually set a synchronizationMechanism? I can totally see how users would forget to do this.

It also bothers me that you have this public setter on Remixer which is a singleton. What happens if multiple parts of the user's app calls setSynchronizationMechanism()?